presets/ragengine/vector_store/base.py

```diff
@@ -381,20 +422,21 @@ async def chat_completion(self, request: dict) -> ChatCompletionResponse:
         logger.info(
             f"Creating chat engine for index '{request.get('index_name')}' with prompt size: {prompt_len}"
         )
-        chat_engine = self.index_map[
-            request.get("index_name")
-        ].as_chat_engine(
+        # Calculate top_k based on available context. For larger windows, we can afford to retrieve more documents.
+        # 750 tokens is a rough estimate based off max 1500 for code and 1000 for text
```

Comment:
Can you elaborate on how to derive 750 from 1500 and 1000?

Reply:
Added a env var for this and added comments on how it was calculated, also changed to 500.